flash/core/data/io/input.py

```diff
@@ -300,43 +300,6 @@ def predict_load_sample(self, sample: Dict[str, Any]) -> Any:
         """
         return self.load_sample(sample)
 
-    def __getstate__(self):
-        """Temporarily override pickle behaviour.
-
-        TODO: New DataPipeline should avoid this being pickled.
-        """
-        state = self.__dict__.copy()
-        state.pop("data")
-        if "data_iter" in state:
-            state.pop("data_iter")
-        return state
-
-    def __setstate__(self, newstate):
-        """Temporarily override pickle behaviour.
-
-        TODO: New DataPipeline should avoid this being pickled.
-        """
-        newstate["data"] = None
-        self.__dict__.update(newstate)
-
-    def __copy__(self):
-        """The default copy implementation seems to use ``__getstate__`` and ``__setstate__`` so we override it
-        here with a custom implementation to ensure that it includes the data list."""
-        cls = self.__class__
-        result = cls.__new__(cls)
-        result.__dict__.update(self.__dict__)
-        return result
-
-    def __deepcopy__(self, memo):
-        """The default deepcopy implementation seems to use ``__getstate__`` and ``__setstate__`` so we override it
-        here with a custom implementation to ensure that it includes the data list."""
-        cls = self.__class__
-        result = cls.__new__(cls)
-        memo[id(self)] = result
-        for k, v in self.__dict__.items():
-            setattr(result, k, deepcopy(v, memo))
-        return result
```

Comment:
@krshrimali This is the main fix. We used to have a bug where the data was accidentally included in the checkpoint. We patched that by adding this overrides. But then DDP spawn needs to pickle the data to send it to each process so this causes problems. We refactored away the bit that got this included in the checkpoint so now can be safely removed :smiley:

Reply:
Awesome, thank you so much for the explanation, @ethanwharris!